src/pool/srv_util.c

```diff
@@ -348,8 +271,7 @@ ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
 	 * in the pool map and not present in the list of replicas.
 	 **/
 	for (i = 0, nreplaced = 0; i < nnodes && nreplaced < nfailed; i++) {
-		if (!map_ranks_include(MAP_RANKS_UP,
-				       nodes[i].do_comp.co_status))
+		if ((PO_COMP_ST_DOWNOUT | PO_COMP_ST_DOWN) & nodes[i].do_comp.co_status)
```

Comment:
[Question] Do you think we should also skip DRAIN ranks here, that is, to avoid add a draining rank as a PS new replicas? (I'm asking because I have a pull request that changes ds_pool_check_failed_replicas too, and I will have to address the conflicts with this pull request. I wonder if I should add DRAIN to the condition here.)

Reply:
> Do you think we should also skip DRAIN ranks here, that is, to avoid add a draining rank as a PS new replicas?

yes, I think it makes sense, but the function name is a bit confusing, since drain ranks are still available targets from the view of I/O.  just in case some one might re-use this this function for other purpose.